### PR TITLE
fix stray Global Tags post GeometricDetExtra Migration

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

After PR #31729 has been integrated, there have been falilures in 20 unit tests in `DQM/Integration` and exceptions in wf 1001.2, 1030.0, 1040.0, 1040.1 (as reported in issue:  #31869)

This PR fixes that state of affairs, by doing two things:
* fixes some stray GTs left around in relval tests and migrate them with the autoCond key: `auto:run3_data_express`
* updates the `DQM/Integration` code to use the new Express Reco GT: `111X_dataRun3_Express_v4`

#### PR validation:

I've run successfully :
   * unit tests: `scramv1 b runtests`
   * relval tests: `runTheMatrix.py -l 1001.2,1030.0,1040.0,1040.1 -t 4 -j 8`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

This PR is not a backport, no backport is needed.

### CODE REVIEW

File: DQM/Integration/python/config/FrontierCondition_GT_cfi.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 import FWCore.ParameterSet.Config as cms
 from Configuration.StandardSequences.FrontierConditions_GlobalTag_cff import *
-GlobalTag.globaltag = "111X_dataRun3_HLT_GEMRecoGEO_v1"
+GlobalTag.globaltag = "111X_dataRun3_Express_v4"
```
Comment (mmusich) [inline DQM/Integration/python/config/FrontierCondition_GT_cfi.py:3]: @christopheralanwest, to be fair, a new Run3 HLT Global Tag is needed here (but I haven't found any without the records removed in #31729 .

Comment (christopheralanwest) [reply DQM/Integration/python/config/FrontierCondition_GT_cfi.py:3]: I created a new GT, `111X_dataRun3_HLT_v3`, with the obsolete records removed: https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/111X_dataRun3_HLT_v2/111X_dataRun3_HLT_v3

Comment (mmusich) [reply DQM/Integration/python/config/FrontierCondition_GT_cfi.py]: thanks, fixed now.

**Comment by silviodonato** [comment]
urgent

**Comment by chayanit** [comment]
please test

**Comment by christopheralanwest** [comment]
> I found errors in the following unit tests:
> 
> ---> test TestDQMOnlineClient-beam_dqm_sourceclient had ERRORS

I think that in addition to updating the GT here:

https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py#L76

the hardcoded GT should also be commented out. It looks like the offending line is only intended to show an example of overriding the GT.

**Comment by christopheralanwest** [comment]
test parameters
workflow = 1001.2,1030.0,1040.0,1040.1

**Comment by mrodozov** [comment]
test parameters
- workflow = 1001.2,1030.0,1040.0,1040.1

**Comment by mrodozov** [comment]
I don't remember if it worked without - for each param so lets put it

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py [hunk 2]
```diff
@@ -73,7 +73,7 @@
     from Configuration.AlCa.GlobalTag import GlobalTag as gtCustomise
     process.GlobalTag = gtCustomise(process.GlobalTag, 'auto:run2_data', '')
     # you may need to set manually the GT in the line below
-    process.GlobalTag.globaltag = '100X_upgrade2018_realistic_v10'
+    #process.GlobalTag.globaltag = '100X_upgrade2018_realistic_v10'
```
Comment (mmusich) [inline DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py:76]: actually wondering what was being tested here exactly, by running a "data"-like workflow with a MC Global Tag?!?

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
@cms-sw/pdmv-l2 @cms-sw/upgrade-l2 if there are no objections, I will merge this PR before the night IB

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
merge

**Pull request merged**